pkg/proxy/winkernel/proxier.go

```diff
@@ -188,20 +187,24 @@ func (t DualStackCompatTester) DualStackCompatible(networkName string) bool {
 		return false
 	}
 
-	globals, err := hcn.GetGlobals()
-	if err != nil {
-		klog.ErrorS(err, "Unable to determine networking stack version. Falling back to single-stack")
-		return false
-	}
-
-	if !kernelSupportsDualstack(globals.Version) {
-		klog.InfoS("This version of Windows does not support dual-stack. Falling back to single-stack")
+	// First tag of hcsshim that has a proper check for dual stack support is v0.8.17, although it would've been hard to use that tag due
+	// to some recursive dependency issues in k8s. These issues were fixed in v0.8.22 so that is really the first usable tag for k8s
+	// that has the changes.
+	if err := hcn.IPv6DualStackSupported(); err != nil {
+		// Hcn *can* fail the query to grab the version of hcn itself (which this call will do internally before parsing
+		// to see if dual stack is supported), but the only time this can happen, at least that can be discerned, is if the host
+		// is pre-1803 and hcn didn't exist. hcsshim should truthfully return a known error if this happened that we can
+		// check against, and the case where 'err != this known error' would be the 'this feature isn't supported' case, as is being
+		// used here. For now, seeming as how nothing before ws2019 (1809) is listed as supported for k8s we can pretty much assume
+		// any error here isn't because the query failed, it's just that dualstack simply isn't supported on the host. With all
+		// that in mind, just log as info and not error to let the user know we're falling back.
+		klog.Infof("%s. Falling back to single-stack", err.Error())
```

Comment:
can you use InfoS here?  See the structured logging guide here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions-to-structured-equivalent

Reply:
As well as passing the error as a key: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#using-errors

I think the Message should probably stay something like `This version of Windows does not support dual-stack. Falling back to single-stack` and have an error key

Reply:
Done